src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ImportSectionNode.cs

```diff
@@ -135,8 +130,10 @@ public override void EncodeData(ref ObjectDataBuilder dataBuilder, NodeFactory f
                 dataBuilder.EmitUInt(0);
             }
 
-            if (_emitGCRefMap)
+            // If we emit an Rva for an empty gcrefmap, it will have no size header and be impossible for r2rdump to read correctly
+            if (_gcRefMap?.IsEmpty == false)
```

Comment:
```suggestion
            if (_gcRefMap != null && !_gcRefMap.IsEmpty)
```
Actually, I do not think the shortcut is going to work correctly here.

Reply:
Ok, I think it does work correctly. Ignore this comment.